src/components/SelectionDetails.tsx

```diff
@@ -12,12 +12,12 @@ const mapStateToProps = (state: DiagramState): SelectionDetailsStateProps => {
     };
 };
 
-const SelectionDetails = ({ nodes }: SelectionDetailsStateProps) => {
-    const message = nodes.reduce((result: string, current: string) => result + ' ' + current, '');
-    return (
-        <div>{nodes.length === 0 ? 'No selection' : 'Selection: ' + message}</div>
-    );
-};
+class SelectionDetails extends React.Component<SelectionDetailsStateProps, {}> {
+    message = this.props.nodes.reduce((result: string, current: string) => result + ' ' + current, '');
```

Comment:
I think it is not working because with this change, the message is not computed on each render call (so the displayed message is not updated when this.props.nodes changes)